Modules/Devices/src/Devices.Application/PushNotifications/Commands/UpdateDeviceRegistration/Handler.cs

```diff
@@ -22,10 +22,14 @@ public Handler(IPushService pushService, IUserContext userContext)
 
     public async Task<Unit> Handle(UpdateDeviceRegistrationCommand request, CancellationToken cancellationToken)
     {
-        var handle = PnsHandle.Parse(request.Handle, DeserializePlatform(request.Platform));
-        if (handle.IsSuccess)
+        var parseHandleResult = PnsHandle.Parse(request.Handle, DeserializePlatform(request.Platform));
+        if (parseHandleResult.IsSuccess)
         {
-            await _pushService.UpdateRegistration(_activeIdentity, _activeDevice, handle.Value, cancellationToken);
+            await _pushService.UpdateRegistration(_activeIdentity, _activeDevice, parseHandleResult.Value, cancellationToken);
+        }
+        else
+        {
+            throw new ArgumentException(parseHandleResult.Error.Message);
```

Comment:
Don't throw an ArgumentException. This will result in a 500. But we want a 400. So you should throw an DomainException.